poc-data-api/Extracting_Water_Temperature_at_Site.Rmd

```diff
@@ -0,0 +1,251 @@
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# Goal of this notebook
+
+This notebook will demonstrate how to access the RIMReP `geoparquet` collection for AIMS Sea Surface Temperature Monitoring Program. This dataset includes data since 1980 for tropical and subtropical coral reefs around Australia, including 80 Great Barrier Reef sites.
+
+Then, we will extract the coordinates for all sites sampled in this monitoring program. This way we can extract data for the site of our choice using the name of the site of our interest, without the need to know their coordinates.
+
+# Loading libraries
+
+```{r libraries, results = "hide", message = F, warning = F}
+library(arrow)
+library(dplyr)
+library(magrittr)
+library(stringr)
+library(wkb)
+library(ggplot2)
+library(sf)
+library(lubridate)
+```
+
+# Connecting to RIMReP collection
+
+As mentioned above, we will connect to the AIMS Sea Surface Temperature Monitoring Program. This can take a minute or so.
+
+```{r connect}
+#Establishing connection
+data_bucket <- s3_bucket("s3://rimrep-data-public/091-aims-sst/test-50-64-spatialpart")
+
+#Accessing dataset
+data_df <- open_dataset(data_bucket)
+```
+
+## Exploring dataset structure
+
+We can explore the dataset `schema` to understand its structure, and identify the fields that are relevant to us.
+
+```{r explore}
+#Checking data structure
+data_df$schema
+```
+We can see that there are a number of variables available in this dataset. We will need to access three variables to create a list of all sampled sites in this monitoring program:  
+- `site`, which includes the name of all sites where temperature loggers were deployed
+- `subsite`, which includes a shorten version of the site name and an indication of where loggers where deployed (see below for more details)
+- `geometry`, which includes latitude and longitude coordinates in [well-known binary (WKB)](https://loc.gov/preservation/digital/formats/fdd/fdd000549.shtml) format
+  
+Based on the `subsite` information, we will create a new column that we will call `deployment_location`. If `subsite` string includes `SL`, this indicates the temperature logger was deployed on a reef slope (depths of 3 m or more). If this is the case, our new column will be labelled `reef slope`. If `FL` is included in the `subsite` string, this means the temperature logger was deployed on a reef flat (areas less than 3 m deep). In this case, we will classify this as `reef flat` in our new column.  
+  
+Next, we will transform the information contained in the `geometry` field into coordinate pairs (i.e., latitude and longitude in decimal degrees).  
+  
+**Note** that you can get additional information about the AIMS dataset by exploring its metadata as shown below.
+
+```{r metadata}
+#Checking metadata
+data_df$metadata
+```
+# Extracting variables of our interest from dataset
+
+We can extract data from the AIMS dataset by using `dplyr` verbs as shown below.

```

Comment:
We can extract location (coordinates) data from the AIMS dataset by using `dplyr` verbs as shown below.